### PR TITLE
pub date falls back to rss_fetcher date when missing- skipping mypy p…

### PR DESCRIPTION
**Pull request by pgulley**
…recommit due to local dependency issue

Dev instance of the pipeline still runs without errors

### CODE REVIEW

**Comment by pgulley** [comment]
For #186 and #172

File: indexer/workers/importer.py [hunk 1]
```diff
@@ -158,14 +158,21 @@ def process_story(self, sender: StorySender, story: BaseStory) -> None:
...
 
             keys_to_skip = ["is_homepage", "is_shortened"]
+
             data: Mapping[str, Optional[Union[str, bool]]] = {
                 k: v for k, v in content_metadata.items() if k not in keys_to_skip
             }
+
+            # if publication date is none, fallback to rss_fetcher pub_date
+            if data["publication_date"] is None:
+                data["publication_date"] = story.rss_entry().as_dict()["pub_date"]
```
Comment (kilemensi) [inline indexer/workers/importer.py:168]: Doesn't `story.rss_entry().pub_date` return the `pub_date` without converting everything to `dict` first?

File: indexer/workers/importer.py [hunk 2]
```diff
@@ -177,7 +184,11 @@ def import_story(
         if data:
             url = str(data.get("url"))
             url_hash = hashlib.sha256(url.encode("utf-8")).hexdigest()
-            publication_date = str(data.get("publication_date"))
+            # We want actual None, not 'None', if pub_date is missing
+            if (pub_date := data.get("publication_date")) is None:
+                publication_date = None
+            else:
+                publication_date = str(pub_date)
```
Comment (kilemensi) [inline indexer/workers/importer.py:191]:
```suggestion
            # We want actual None, not 'None', if publication_date is missing
            if ("publication_date" in data) is None:
                publication_date = str(data["publication_date"])
            else:
                publication_date = None
```

**Review by kilemensi — COMMENTED** [review]
👍🏽 

--

Shouldn't this PR address [this comment](https://github.com/mediacloud/story-indexer/blob/6f90ae7fb2467d1ea4d820b1c78db3a706a27124/indexer/workers/parser.py#L123) as well?

File: indexer/workers/importer.py [hunk 3]
```diff
@@ -185,7 +184,7 @@ def import_story(
             url = str(data.get("url"))
             url_hash = hashlib.sha256(url.encode("utf-8")).hexdigest()
             # We want actual None, not 'None', if publication_date is missing
-            if ("publication_date" in data):
+            if "publication_date" in data:
```
Comment (philbudne) [inline indexer/workers/importer.py:187]: Can "publication_date" item be present, but `None` or `"None"` (especially when replaying WARC files)??

Comment (pgulley) [reply indexer/workers/importer.py]: Good point, i've addressed this in the recent commit

**Review by philbudne — APPROVED** [review]
Looks good.  `data.get("publication_date") not in [None, "None"]` would be shorter

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**